File: server/models/Article.ts
import mongoose from 'mongoose';

const articleSchema = new mongoose.Schema({
    title: {
        type: String,
        required: true,
        trim: true
    },
    slug: {
        type: String,
        required: true,
        unique: true,
        lowercase: true
    },
    excerpt: {
        type: String,
        required: true
    },
    content: {
        type: Object, // Tiptap JSON content
        required: true
    },
    coverImage: {
        type: String,
        default: ''
    },
    published: {
        type: Boolean,
        default: false
    },
    publishedAt: {
        type: Date
    },
    author: {
        type: String,
        default: 'Yusuf Akinleye'
    }
}, {
    timestamps: true
});

export const Article = mongoose.models.Article || mongoose.model('Article', articleSchema);

File: backend/internal/models/article.go
package models

import (
	"time"

	"go.mongodb.org/mongo-driver/v2/bson"
)

type Article struct {
	ID          bson.ObjectID `bson:"_id,omitempty" json:"id"`
	Title       string        `bson:"title" json:"title"`
	Slug        string        `bson:"slug" json:"slug"`
	Excerpt     string        `bson:"excerpt" json:"excerpt"`
	Content     string        `bson:"content" json:"content"` // Stored as JSON string or plain text
	CoverImage  string        `bson:"cover_image" json:"coverImage"`
	Published   bool          `bson:"published" json:"published"`
	PublishedAt *time.Time    `bson:"published_at" json:"publishedAt"`
	Author      string        `bson:"author" json:"author"`
	Views       int           `bson:"views" json:"views"`
	CreatedAt   time.Time     `bson:"created_at" json:"createdAt"`
	UpdatedAt   time.Time     `bson:"updated_at" json:"updatedAt"`
}
